ocs_ci/utility/multicluster.py

```diff
@@ -73,23 +64,43 @@ def generate_zone_ranks(self):
             )
         log.info(f"zone ranks = {self.zone_ranks}")
 
-    @abstractmethod
-    def generate_role_ranks(self):
+    def generate_config_index_map(self):
         """
-        Based on the multicluster scenario, child class should generate the corresponding
-        role ranks. Roles are specific to multicluster scenarios
+        Generate config indexes for all the MDRs/RDR cluster roles
+        ex: {"ActiveACM": 0, "PassiveACM": 2, "PrimaryODF": 1, "SecondaryODF": 3}
 
         """
-        pass
+        for cluster in ocsci_config.clusters:
+            cluster_index = cluster.MULTICLUSTER["multicluster_index"]
+            if cluster_index == get_active_acm_index():
+                self.roles_to_config_index_map["ActiveACM"] = cluster_index
+            elif cluster_index == get_primary_cluster_index():
+                self.roles_to_config_index_map["PrimaryODF"] = cluster_index
+            elif (
+                cluster_index in get_all_acm_indexes()
+                and cluster_index != get_active_acm_index()
```

Comment:
There seems to be also `get_passive_acm_index` function in the `ocs_ci.ocs.utils` module, which will make this condition little bit cleaner (similar to the previous two).

Reply:
done